server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -982,6 +983,127 @@ public void onAckTimeout() {
         }
     }
 
+    @TestLogging(value = "org.elasticsearch.cluster.service.MasterService:WARN", reason = "testing WARN logging")
+    public void testStarvationLogging() throws Exception {
+        final long warnThresholdMillis = MasterService.MASTER_SERVICE_STARVATION_LOGGING_THRESHOLD_SETTING.get(Settings.EMPTY).millis();
+        relativeTimeInMillis = randomLongBetween(0, Long.MAX_VALUE - warnThresholdMillis * 3);
+        final long startTimeMillis = relativeTimeInMillis;
+        final long taskDurationMillis = TimeValue.timeValueSeconds(1).millis();
+
+        MockLogAppender mockAppender = new MockLogAppender();
+        mockAppender.start();
+
+        Logger clusterLogger = LogManager.getLogger(MasterService.class);
+        Loggers.addAppender(clusterLogger, mockAppender);
+        try (MasterService masterService = createMasterService(true)) {
+            final AtomicBoolean keepRunning = new AtomicBoolean(true);
+
+            final Runnable await = new Runnable() {
+                private final CyclicBarrier cyclicBarrier = new CyclicBarrier(2);
+
+                @Override
+                public void run() {
+                    try {
+                        cyclicBarrier.await(5, TimeUnit.SECONDS);
```

Comment:
I would prefer to stick to our "proven" 10 second value here and below.

Reply:
Ok, changed in 4092bf38e425b7596a5e9ad44427fba522ad9e90.